src/SymbolCollector.Core/Client.cs

```diff
@@ -30,59 +30,75 @@ public Client(
         BatchType type,
         IEnumerable<string> topLevelPaths,
+        ISpan span,
         CancellationToken cancellationToken)
     {
-        var groupsSpan = SentrySdk.GetSpan()?.StartChild("group.get", "Get the group of directories to search in parallel");
+        List<List<string>> groups;
         var counter = 0;
-        var groups =
-            (from topPath in topLevelPaths
-                from lookupDirectory in SafeGetDirectories(topPath)
-                where _blockListedPaths?.Contains(lookupDirectory) != true
-                let c = counter++
-                group lookupDirectory by c / ParallelTasks
-                into grp
-                select grp.ToList()).ToList();
-        groupsSpan?.Finish();
-
-        var startSpan = SentrySdk.GetSpan()?.StartChild("batch.start");
-        Guid batchId;
-        try
         {
-            batchId = await _symbolClient.Start(friendlyName, type, cancellationToken);
-            startSpan?.Finish(SpanStatus.Ok);
+            var groupsSpan = span.StartChild("group.get", "Get the group of directories to search in parallel");
+            groups =
+                (from topPath in topLevelPaths
+                    from lookupDirectory in SafeGetDirectories(topPath)
+                    where _blockListedPaths?.Contains(lookupDirectory) != true
+                    let c = counter++
+                    group lookupDirectory by c / ParallelTasks
+                    into grp
+                    select grp.ToList()).ToList();
+            groupsSpan.Finish();
         }
-        catch (Exception e)
+
+        Guid batchId;
         {
-            startSpan?.Finish(e);
-            throw;
+            var startSpan = span.StartChild("batch.start");
+            try
+            {
+                batchId = await _symbolClient.Start(friendlyName, type, cancellationToken);
+                startSpan.Finish(SpanStatus.Ok);
+            }
+            catch (Exception e)
+            {
+                startSpan.Finish(e);
+                throw;
+            }
         }
 
-        var uploadSpan = SentrySdk.GetSpan()?.StartChild("batch.upload");
-        uploadSpan?.SetTag("groups", groups.Count.ToString());
-        uploadSpan?.SetTag("total_items", counter.ToString());
-        try
         {
-            foreach (var group in groups)
+            var uploadSpan = span.StartChild("batch.upload", "concurrent batch upload");
+            uploadSpan.SetData("groups", groups.Count.ToString());
+            uploadSpan.SetData("total_items", counter.ToString());
+
+            // use this as parent to all outgoing HTTP requests now:
+            SentrySdk.ConfigureScope(s => s.Span = uploadSpan);
+            int i = 0;
+            try
             {
-                await UploadParallel(batchId, group, cancellationToken);
+                foreach (var group in groups)
+                {
+                    if (i++ == 2) break;
```

Comment:
TODO: Delete this
```suggestion
```

Only running the first 50 (1 group) because if I run 2, the transaction never shows up in Sentry
Need to find out why